src/ruvsarpur.py

```diff
@@ -256,13 +263,16 @@ def parseArguments():
  
 # Saves a list of program ids to a file
 def savePreviouslyRecordedShows(pids):
-  with open('prevrecorded.log', 'w+') as thefile:
+  #make sure that the directory exists
+  os.makedirs(os.path.dirname(PREV_LOG_FILE), exist_ok=True)
```

Comment:
There is a minor issue with the exist_ok parameter in earlier Python 3 versions. It might be safer to do an explicit if dir does not exist check here instead of suppressing the error?